gpflow/ekernels.py

```diff
@@ -138,7 +138,12 @@ def eKzxKxz(self, Z, Xmu, Xcov):
         D = tf.shape(Xmu)[1]
         lengthscales = self.lengthscales if self.ARD else tf.zeros((D,), dtype=settings.float_type) + self.lengthscales
 
-        Kmms = tf.sqrt(self.K(Z, presliced=True)) / self.variance ** 0.5
+        # If an element of Kmms is 0, then its gradient is +inf and that gets
+        # backpropagated, but the overall gradient should instead be 0 (because
+        # the overall function is exp(-1/2 x). We fix this with `tf.where`.
+        Kmms_sq = self.K(Z, presliced=True)
+        Kmms = tf.sqrt(tf.where(tf.equal(Kmms_sq, tf.zeros((), dtype=settings.float_type)),
```

Comment:
Can it be different from `settings.float_type` type? If it is even not, it still should be `Kmms_sq.dtype`.

Reply:
I think you're correct, it should be `Kmms_sq.dtype`.